app/models/host.rb

```diff
@@ -1859,6 +1859,10 @@ def read_only_storages
     storages.where(:host_storages => {:read_only => true})
   end
 
+  def available_storages
+    writable_storages.reject(&:maintenance)
```

Comment:
> perhaps we should do so in a comprehensive manner

Okay do you want to do that before the datastore maintenance stuff or after?

And just so I understand, we currently have `MiqRequestWorkflow#allowed_storages` which filters storages based on object selection (like host), we would add a `Storage::provisionable_storages` which would `allowed_storages` would call instead of the `host.writable_storages` that it does now?